uoyweek.cpp

```diff
@@ -83,19 +101,19 @@ Term * getTerm(time_t now) {
 string getString(bool capitalize) {
     time_t now = time(0);
     Term *term = getTerm(now);
-	string code = term->code;
+    string code = term->code;
     int week = getWeek(now, term);
-    string day = days[gmtime(&now)->tm_wday].substr(0,3);
-	if (capitalize) {
-		code[0] = std::toupper(code[0]);
-		day[0] = std::toupper(day[0]);
-	}
-    int length = strlen(term->code.c_str()) + 2 + 3 + (week>=10?2:1) + 1;
-    char* str = (char*) malloc(sizeof(char) * length);
+    string day = DAYS[gmtime(&now)->tm_wday].substr(0,3);
+    if (capitalize) {
+        code[0] = std::toupper(code[0]);
+        day[0] = std::toupper(day[0]);
+    }
+    int length = strlen(term->code.c_str()) + 2 + 3 + (week/10 + 1) + 1;
+    char *str = (char *)malloc(sizeof(char) * length);
```

Comment:
good lord no. not in C++. `new` at the very least, or even better, a stringstream

Reply:
I'll look into stringstreams